server/protocols/protocols-imap4/src/main/java/org/apache/james/imapserver/netty/ImapChannelUpstreamHandler.java

```diff
@@ -173,6 +184,7 @@ public void channelActive(ChannelHandlerContext ctx) throws Exception {
         ImapSession imapsession = new NettyImapSession(ctx.channel(), secure, compress, authenticationConfiguration.isSSLRequired(),
             authenticationConfiguration.isPlainAuthEnabled(), sessionId,
             authenticationConfiguration.getOidcSASLConfiguration());
+        connectionChecks.forEach(connectionCheck -> Mono.from(connectionCheck.validate(imapsession.getRemoteAddress())).block());

```

Comment:
This is call is useless at best, extermely dangerous at worst if HA proxy is enabled...
channelActive is called of course also when establishing a connection when proxy protocol is on, then called again after receiving the proxy protocol payload.
Please do not call this here if the proxy protocol is on.